static/app/views/replays/detail/breadcrumbs/useBreadcrumbFilters.tsx

```diff
@@ -0,0 +1,152 @@
+import {useCallback, useMemo} from 'react';
+import uniq from 'lodash/uniq';
+
+import {decodeList, decodeScalar} from 'sentry/utils/queryString';
+import useFiltersInLocationQuery from 'sentry/utils/replays/hooks/useFiltersInLocationQuery';
+import {getFrameOpOrCategory, ReplayFrame} from 'sentry/utils/replays/types';
+import {filterItems} from 'sentry/views/replays/detail/utils';
+
+export type FilterFields = {
+  f_b_search: string;
+  f_b_type: string[];
+};
+
+type Options = {
+  frames: ReplayFrame[];
+};
+
+type Return = {
+  getBreadcrumbTypes: () => {label: string; value: string}[];
+  items: ReplayFrame[];
+  searchTerm: string;
+  setSearchTerm: (searchTerm: string) => void;
+  setType: (type: string[]) => void;
+  type: string[];
+};
+
+const OPORCATEGORY_TO_TYPE: Record<string, string> = {
+  ['replay.init']: 'start',
+  ['replay.mutations']: 'replay',
+  ['issue']: 'issue',
+  ['console']: 'console',
+  ['navigation']: 'nav',
+  ['navigation.push']: 'nav',
+  ['navigation.navigate']: 'pageLoad',
+  ['navigation.reload']: 'reload',
+  ['navigation.back_forward']: 'navBackForward',
+  ['memory']: 'memory',
+  ['paint']: 'paint',
+  ['ui.blur']: 'blur',
+  ['ui.focus']: 'action',
+  ['ui.multiClick']: 'rageOrMulti',
+  ['ui.slowClickDetected']: 'rageOrDead',
+  ['largest-contentful-paint']: 'lcp',
+  ['ui.click']: 'click',
+  ['ui.keyDown']: 'keydown',
+  ['ui.input']: 'input',
```

Comment:
nit, but the square braces here are not needed.

Square braces on the left help in cases where the key is inside a variable ie:
```
const nameField = 'display_name';
const me = {
  [nameField]: 'ryan',
};
console.log(me); // returns `{display_name: 'ryan'}`
```

```suggestion
  'replay.init': 'start',
  'replay.mutations': 'replay',
  'issue': 'issue',
  'console': 'console',
  'navigation': 'nav',
  'navigation.push': 'nav',
  'navigation.navigate': 'pageLoad',
  'navigation.reload': 'reload',
  'navigation.back_forward': 'navBackForward',
  'memory': 'memory',
  'paint': 'paint',
  'ui.blur': 'blur',
  'ui.focus': 'action',
  'ui.multiClick': 'rageOrMulti',
  'ui.slowClickDetected': 'rageOrDead',
  'largest-contentful-paint': 'lcp',
  'ui.click': 'click',
  'ui.keyDown': 'keydown',
  'ui.input': 'input',
```